src/python/dxpy/workflow_builder.py

```diff
@@ -108,91 +110,275 @@ def _get_unsupported_keys(keys, supported_keys):
+
+
 def _get_validated_json(json_spec, args):
     """
-    Validates dxworkflow.json and returns the json that can be sent with the /workflow/new API request.
+    Validates dxworkflow.json and returns the json that can be sent with the
+    /workflow/new API or /globalworkflow/new request.
     """
     if not json_spec:
         return
     if not args:
         return
 
     # print ignored keys if present in json_spec
-    supported_keys = set(["project", "folder", "name", "outputFolder", "stages",
-                          "inputs", "outputs"])
+    # TODO: add "regionalOptions" to supported_keys when building multi-region workflows is enabled
+    supported_keys = {"project", "folder", "name", "outputFolder", "stages",
+                      "inputs", "outputs", "version", "title", "summary", "categories",
+                      "dxapi"}
     unsupported_keys = _get_unsupported_keys(json_spec.keys(), supported_keys)
     if len(unsupported_keys) > 0:
         print("Warning: the following root level fields are not supported and will be ignored: {}"
               .format(", ".join(unsupported_keys)))
 
     dxpy.executable_builder.inline_documentation_files(json_spec, args.src_dir)
 
-    override_project_id, override_folder, override_workflow_name = \
-        dxpy.executable_builder.get_parsed_destination(args.destination)
-    json_spec['project'] = _get_destination_project(json_spec, args, override_project_id)
-    json_spec['folder'] = _get_destination_folder(json_spec, override_folder)
-
-    workflow_name = _get_workflow_name(json_spec, override_workflow_name)
-    if not workflow_name:
-        print('Warning: workflow name is not specified')
-    else:
-        json_spec['name'] = workflow_name
-
     if 'stages' in json_spec:
         json_spec['stages'] = _get_validated_stages(json_spec['stages'])
 
+    if 'name' in json_spec:
+        if args.src_dir != json_spec['name']:
+            logger.warn(
+                'workflow name "%s" does not match containing directory "%s"' % (json_spec['name'], args.src_dir))
+
+    # Project-based workflow specific validation
+    if args.mode == 'workflow':
+        validated = _validate_json_for_regular_workflow(json_spec, args)
+        json_spec.update(validated)
+
+    # Global workflow specific validation
+    if args.mode == 'globalworkflow':
+        _validate_json_for_global_workflow(json_spec, args)
+
     return json_spec
 
 
-def _create_workflow(json_spec):
+def _build_regular_workflow(json_spec):
     """
-    Creates a closed workflow on the platform.
-    Returns a workflow_id, or None if the workflow cannot be created.
+    Precondition: json_spec must be validated
     """
+    workflow_id = dxpy.api.workflow_new(json_spec)["id"]
+    dxpy.api.workflow_close(workflow_id)
+    return workflow_id
+
+
+def _build_underlying_workflows(json_spec, args):
+    """
+    Creates a workflow in a temporary project for each enabled region.
```

Comment:
Minor: I would put the TODO below as a part of this documentation.  Specifically, the documentation would say that it will build the workflow in the region of the current project but will be expanded in the future to build in all regions.